packages/angular_devkit/core/src/json/schema/registry.ts

```diff
@@ -126,22 +126,27 @@ export class CoreSchemaRegistry implements SchemaRegistry {
     this._ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'));
   }
 
-  private _fetch(uri: string): Promise<JsonObject> {
+  private async _fetch(uri: string): Promise<JsonObject> {
     const maybeSchema = this._uriCache.get(uri);
 
     if (maybeSchema) {
       return Promise.resolve(maybeSchema);
     }
 
     // Try all handlers, one after the other.
-    for (const maybeHandler of this._uriHandlers) {
-      const handler = maybeHandler(uri);
-      if (handler) {
-        // The AJV API only understands Promises.
-        return from(handler).pipe(
-          tap(json => this._uriCache.set(uri, json)),
-        ).toPromise();
+    for (const handler of this._uriHandlers) {
+      let value = await handler(uri);
+      if (value === null || value === undefined) {
+        continue;
       }
+
+      if (isObservable(value)) {
+        value = await value.toPromise();
+      }
```

Comment:
`let value = await handler(uri);` will cause loose of type info on `value`.

I think something like the below is clearer and retains types.

```suggestion
      let value = handler(uri);
      if (!value) {
        continue;
      }

      if (isObservable(value)) {
        value = value.toPromise();
      }

      value = await value;
```